### PR TITLE
To `polkadot-v0.9.13`

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
```
error: cannot find macro `asm` in this scope
  --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:57:5
   |
57 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm
```

In order to make this compilable I have to cherrypick https://github.com/paritytech/substrate/pull/10674.

AFAK, rsix is quite an old package. And the wasmtime drop this dep in the later version.

**Comment by AurevoirXavier** [comment]
```
  error[E0277]: the trait bound `fp_self_contained::UncheckedExtrinsic<MultiAddress<AccountId32, ()>, Call, MultiSignature, (CheckSpecVersion<Runtime>, CheckTxVersion<Runtime>, CheckGenesis<Runtime>, CheckEra<Runtime>, CheckNonce<Runtime>, CheckWeight<Runtime>, ChargeTransactionPayment<Runtime>, CheckEthereumRelayHeaderParcel<Runtime>)>: TypeInfo` is not satisfied
     --> /root/code/darwinia-network/common/node/runtime/pangolin/src/lib.rs:147:24
      |
  147 |         UncheckedExtrinsic = UncheckedExtrinsic
      |                              ^^^^^^^^^^^^^^^^^^ the trait `TypeInfo` is not implemented for `fp_self_contained::UncheckedExtrinsic<MultiAddress<AccountId32, ()>, Call, MultiSignature, (CheckSpecVersion<Runtime>, CheckTxVersion<Runtime>, CheckGenesis<Runtime>, CheckEra<Runtime>, CheckNonce<Runtime>, CheckWeight<Runtime>, ChargeTransactionPayment<Runtime>, CheckEthereumRelayHeaderParcel<Runtime>)>`
      |
      = help: the following other types implement trait `TypeInfo`:
                &T
                &mut T
                ()
                (A, B)
                (A, B, C)
                (A, B, C, D)
                (A, B, C, D, E)
                (A, B, C, D, E, F)
              and 537 others
  note: required by a bound in `meta_type`
     --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/scale-info-1.0.0/src/lib.rs:374:17
      |
  374 |     T: ?Sized + TypeInfo + 'static,
      |                 ^^^^^^^^ required by this bound in `meta_type`

  For more information about this error, try `rustc --explain E0277`.
  error: could not compile `pangolin-runtime` due to previous error
```

This is a question for you @AsceticBear.

**Comment by AurevoirXavier** [comment]
I found the `scale-info` version is mismatched.

**Comment by AsceticBear** [comment]
> I found the `scale-info` version is mismatched. All the deps must meet these requirements `scale < 2.0`, `codec < 3.0`.

Fixed. Keep going. You can skip the section related to the dvm if necessary.

**Review by AsceticBear — APPROVED** [review]

File: frame/staking/src/lib.rs [hunk 1]
```diff
@@ -2142,42 +2144,47 @@ pub mod pallet {
...
-		/// - Reads: Stash Account, Bonded, Slashing Spans, Locks
-		/// - Writes: Bonded, Slashing Spans (if S > 0), Ledger, Payee, Validators, Nominators,
-		///   Stash Account, Locks
-		/// - Writes Each: SpanSlash * S
-		/// # </weight>
+		/// It can be called by anyone, as long as `stash` meets the above requirements.
+		///
+		/// Refunds the transaction fees upon successful execution.
 		#[pallet::weight(T::WeightInfo::reap_stash(*num_slashing_spans))]
 		pub fn reap_stash(
```
Comment (AurevoirXavier) [inline frame/staking/src/lib.rs:2160]: Double-check this change.

Comment (AsceticBear) [reply frame/staking/src/lib.rs:2160]: @HackFisher

Comment (HackFisher) [reply frame/staking/src/lib.rs:2160]: LGTM

**Review by HackFisher — APPROVED** [review]

**Pull request merged**